src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -314,7 +328,7 @@ private boolean checkAndAddNewLogEvent(AtomicInteger totalBytesRead,
         if (currentLogLevel.toInt() < desiredLogLevel.toInt()) {
             return false;
         }
-        return addNewLogEvent(totalBytesRead, attemptLogInformation, data.toString());
+        return addNewLogEvent(totalBytesRead, attemptLogInformation, data.toString(), logMessage.getTimestamp());
```

Comment:
What if the timestamp isn't present? Is that possible?

Reply:
It should not be possible since we don't set that the timestamp field is optional. So if it is not there, the deserialization should fail.